### ISSUE TITLE
Sql.MakeDateTime generates different sql

### ISSUE THREAD

**Issue by HemidAbbasov** [linq2db/linq2db#969]
Hi. This is what I'm testing to see the generating SQL's.

```c# 
[Test]
public void MakeDateTime()
{
    using ( var db = new OrclDB() )
    {
        int year = 2018;
        int month = 1;

        var q = from t in db.Masters
                     where t.Id == 1
                     select new
                               {
                                   Date1 = Sql.MakeDateTime(2018, 1    , 1),
                                   Date2 = Sql.MakeDateTime(year, month, 1)
                               };
        q.ToList().ForEach(d => Console.WriteLine(d));
    }
}
```

Generated SQL is:

```sql
SELECT
        TO_DATE('2018-01-01', 'YYYY-MM-DD') as c1,
        To_Timestamp('2018-1-1', 'YYYY-MM-DD HH24:MI:SS') as c2,
FROM
        MASTER t1 
WHERE
        t1.ID = 1
```
Values are the same, but SQL different. Why so?

### Environment details
linq2db version: *1.10.0*  
Database Server: *Oracle 11.2.0.1.0*  
Database Provider: *Oracle.ManagedDataAccess 12.2.1100*  
Operating system: *Windows 8.1*  
Framework version: *.NET Framework 4.5*

**Comment by sdanyliv** [comment]
It is because how these calls are translated to into internal structures. I have corrected code to generate more closer SQL.

**Comment by HemidAbbasov** [comment]
Thanks.

**Comment by sdanyliv** [comment]
Fixed in #970
